packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.tsx

```diff
@@ -312,9 +312,15 @@ function DocSearch({externalUrlRegex, ...props}: DocSearchV4Props) {
   );
 }
 
-export default function SearchBar(): ReactNode {
+export default function SearchBar(props: Partial<DocSearchV4Props>): ReactNode {
```

Comment:
I made the prop type more permissive so that we can override everything (at your own risk)